src/YOUR_NAME_ex/ex05/bounded_sim.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Kevin Martin Lankut'
+__email__ = 'kela@nmbu.no'
+
+from walker_sim import Walker, Simulation
+import random
+
+
+class BoundedWalker(Walker):
+    def __init__(self, start, home, left_limit, right_limit):
+        super().__init__(start, home)
+        """
+        Initialise the walker
+
+        Arguments
+        ---------
+        start : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        self.start = start
+        self.home = home
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+
+    def bounded_move(self):
+        if self.start != self.left_limit and self.start != self.right_limit:
+            Walker.move(self)
+        elif self.start == self.left_limit:
+            self.start += 1
+            self.steps += 1
+        elif self.start == self.right_limit:
+            self.start -= 1
+            self.steps += 1
+
+        """'If the random generator decides that the walker should move beyond
+        a limit, the move is simply not executed.' I can't think of any way
+        to actually execute move(), and check what the random generator
+        generates to then stop it before it goes on to moving the walker
+        and adding steps. I also don't see the point of adding a new generator
+        to then see if it moves or not, considering that at the end you don't
+        want to see amount of *potential* moves, but amount of *steps*, meaning
+        even if the generator goes past a limit, it doesn't count it, so might
+        as well just start moving in the right direction because eventually
+        it will do that."""
+
+
+class BoundedSimulation(Simulation):
+    def __init__(self, start, home, seed, left_limit, right_limit):
+        super().__init__(start, home, seed)
+        """
+        Initialise the simulation
+
+        Arguments
+        ---------
+        start : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        seed : int
+            Random generator seed
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        self.start = start
+        self.home = home
+        self.seed = seed
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+    
+    def bounded_single_walk(self):
```

Comment:
Part of the point behind inheritance is that you are able to overwrite methods so that they have the same name, give the same type as a result but otherwise function differently. You should do this with both `move()` and  `single_walk()`. That way you would not have to rewrite the `run_simulation()`.